[Analytics] httpRequestsByColoGroups to httpRequestsAdaptiveGroups info (#1170)
cloudflare/cloudflare-docs: products/analytics/src/content/graphql-api/migration-guides/graph-ql-analytics/index.md

---

> title: httpRequestsByColoGroups to httpRequestsAdaptiveGroups

Reviewer: ```suggestion
title: httpRequests1mByColoGroups/httpRequests1dByColoGroups to httpRequestsAdaptiveGroups
```

Author: Committing this change to the page title but not the text that appears in the table of contents to help with formatting

order: 10
---

> # httpRequestsByColoGroups to httpRequestsAdaptiveGroups

Reviewer: ```suggestion
# httpRequests1mByColoGroups/httpRequests1dByColoGroups to httpRequestsAdaptiveGroups
```

> This guide shares considerations when migrating from the deprecated `httpRequestsByColoGroups` GraphQL API node to the `httpRequestsAdaptiveGroups` GraphQL API node.

Reviewer: ```suggestion
This guide shares considerations when migrating from the deprecated GraphQL API
nodes `httpRequests1mByColoGroups`, `httpRequests1dByColoGroups` to the
`httpRequestsAdaptiveGroups` node.
```

> Previously a query that used the `httpRequestsByColoGroups` GraphQL API node to access the number of requests per data center, bandwidth, and unique visitors looked like the following:

Reviewer: ```suggestion
Let's see, how it possible to access the number of requests, unique visitors,
and bandwidth used per data center with the deprecated GraphQL API node
`httpRequests1mByColoGroups`:
```

> ```graphql

Author: @victor-perov / @twalwyn , would appreciate your eyes on the httpRequestsByColoGroups example I added - stole it from the other migration guide, not sure if we can cut anything in here out so it's more similar to what's in the updated example

Reviewer: This is not example for **httpRequests1*ByColoGroups** but for **httpRequests1mGroups**. This node doesn't provide aggregations per data center.

Reviewer: I will share a query and the response

                ],
                "countryMap": [
                  {
                    "bytes": 101033,
                    "clientCountryName": "CA",
                    "requests": 2,
                    "threats": 0
                  },
                  {
                    "bytes": 182366,
                    "clientCountryName": "US",
                    "requests": 2,
                    "threats": 0
                  }
                ],
                "encryptedBytes": 283399,
                "encryptedRequests": 4,
                "ipClassMap": [
                  {
                    "ipType": "monitoringService",
                    "requests": 4
                  }
                ],
                "pageViews": 1,
                "requests": 4,
                "responseStatusMap": [
                  {
                    "edgeResponseStatus": 200,
                    "requests": 4
                  }
                ],
                "threatPathingMap": [],
                "threats": 0
              },
              "uniq": {
                "uniques": 4
              }
            }
          ]
        }
      ]
    }
  },
  "errors": null
}
```

> With the deprecation of the `httpRequestsByColoGroups` GraphQL API node, the query to access the same data (`count`, `sum(edgeResponseBytes)`, and `visits` in the `httpRequestsAdaptiveGroups` GraphQL API node) looks like the following example: 

Reviewer: ```suggestion
With the deprecation of the
`httpRequests1mByColoGroups`/`httpRequest1dByColoGroups` GraphQL API nodes, the
query to access the same data (`requests`, `bytes`) and `visits`, with the
`httpRequestsAdaptiveGroups` node looks like the following example:
```

                "visits": 5230
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 7530,
              "dimensions": {
                "metric": "LHR"
              },
              "sum": {
                "edgeResponseBytes": 46665180,
                "visits": 5220
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 5200,
              "dimensions": {
                "metric": "EZE"
              },
              "sum": {
                "edgeResponseBytes": 32085270,
                "visits": 3360
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 5170,
              "dimensions": {
                "metric": "PTY"
              },
              "sum": {
                "edgeResponseBytes": 31573240,
                "visits": 3220
              }
            }
          ]
        }
      ]
    }
  },
  "errors": null
}
```

This query says:

- Given the indicated zones, `limit`, and time range,

> - Fetch `count`, `sum(edgeResponseBytes)`, and `visits` data center analytics by `coloCode`.

Reviewer: ```suggestion
- Fetch the total number of requests (as `count`), the total amount of response
  bytes (as `edgeResponseBytes` of `sum` object), and the total number of
  `visits` per data center (as `metric` of `dimensions` object).
```

A few points to note:

> - Instead of `requests`, the `httpRequestsAdaptiveGroups` query reports `count`, which indicates the number of requests per data center.

Reviewer: ```suggestion
- Instead of `sum(requests)`, the `httpRequestsAdaptiveGroups` node reports
  `count`, which indicates the same number of requests.
```

> - The `httpRequestsAdaptiveGroups` query reports `sum(edgeResponseBytes)` instead of `bandwidth`. Although `httpRequestsByColoGroups` reported this metric as `bandwidth`, calling it `bandwidth` was not an accurate representation of the returned data. 

Reviewer: ```suggestion
- The `httpRequestsAdaptiveGroups` node reports `sum(edgeResponseBytes)`
  instead of `sum(bytes)`.
```

> - `unique visitors per colocation` is not supported in `httpRequestsAdaptiveGroups`, but the `httpRequestsAdaptiveGroups` API does support `visits`. A visit is defined as a page view that originated from a different website or direct link. Cloudflare checks where the HTTP referer does not match the hostname. One visit can consist of multiple page views.

Reviewer: ```suggestion
- `unique visitors per data center` is not supported with
  `httpRequestsAdaptiveGroups` node, but the node does support `visits`. A
  visit is defined as a page view that originated from a different website or
  direct link. Cloudflare checks where the HTTP referer does not match the
  hostname. One visit can consist of multiple page views.
```

---
title: httpRequestsByColoGroups to httpRequestsAdaptiveGroups
order: 10
---

# httpRequestsByColoGroups to httpRequestsAdaptiveGroups

> This guide shares considerations when migrating from the deprecated `httpRequestsByColoGroups` API to the `httpRequestsAdaptiveGroups` API.

Reviewer 2: ```suggestion
This guide shares considerations when migrating from the deprecated `httpRequests1[m|d]ByColoGroups` GarphQL API nodes to the `httpRequestsAdaptiveGroups` GraphQL API node.
```

> This guide shares considerations when migrating from the deprecated `httpRequestsByColoGroups` API to the `httpRequestsAdaptiveGroups` API.

Reviewer: To be precise, both `httpRequestsByColoGroups` and `httpRequestsAdaptiveGroups` are Nodes of the same API - GraphQL API. I don't think it's correct to say that one is deprecated API and another one - not.

I think would be nice to have wording like:
```
This guide shares considerations when migrating from the deprecated GraphQL
API node `httpRequestsByColoGroups` to the `httpRequestsAdaptiveGroups`.
``` 

P.S. We use a coding standard to keep 80 chars long lines if it comes to a text.

> Below is a query from the Cloudflare dashboard that shows how to access number of requests per colocation, bandwidth, and unique visitors with `httpRequestsAdaptiveGroups`.

Reviewer 2: Should we provide an example of retrieving the same data, but from the `httpRequests1mByColoGroups` GraphQL node? The reason for this suggestion is that I, personally, find it useful to have something to compare to. Put another way, I am proposing the following structure: 
* Outline the deprecated way of retrieving "number of requests per colocation, bandwidth, and unique visitors"
* Outline the replacement
* Highlight the differences

> Below is a query from the Cloudflare dashboard that shows how to access number of requests per colocation, bandwidth, and unique visitors with `httpRequestsAdaptiveGroups`.

Reviewer: I don't think we can benefit from bringing the exact query from the Cloudflare dashboard. We use Apollo client to manage communication with GQL, which adds some extra meta fields we can avoid by providing a clear example.

I would expect to see queries in the docs with a simple, clear purpose, just to make the learning process faster.

```code

> query {

Author: Any help cleaning up this query appreciated - not sure the edits I made are 100% correct.

Reviewer: Yeap, the query looks good to me. Though I would suggest some edits, for example (you could use `graphql` as a style for formatting):

```graphql
{
	viewer {
		zones(filter: {zoneTag: $zoneTag}) {
			series: httpRequestsAdaptiveGroups(
				limit: 5,
				orderBy: [ sum_visits_DESC ],
				filter: {
					datetimeFifteenMinutes_geq: $start
					datetimeFifteenMinutes_lt: $end
				}
			) {
				count
				avg {
					sampleInterval
				}
				sum {
					edgeResponseBytes
					visits
				}
				dimensions {
					metric: coloCode
				}
			}
		}
	}
}
```
So it's cleared version of the existing one, but it emphasizes the filter criteria and shows how we can benefit from **orderBy**. This query will get the top 5 colos by visits.

The example of the response could be as follows (json):
```json
{
  "data": {
    "viewer": {
      "zones": [
        {
          "series": [
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 7700,
              "dimensions": {
                "metric": "AMS"
              },
              "sum": {
                "edgeResponseBytes": 50078910,
                "visits": 5700
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 7300,
              "dimensions": {
                "metric": "CDG"
              },
              "sum": {
                "edgeResponseBytes": 47685220,
                "visits": 5230
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 7530,
              "dimensions": {
                "metric": "LHR"
              },
              "sum": {
                "edgeResponseBytes": 46665180,
                "visits": 5220
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 5200,
              "dimensions": {
                "metric": "EZE"
              },
              "sum": {
                "edgeResponseBytes": 32085270,
                "visits": 3360
              }
            },
            {
              "avg": {
                "sampleInterval": 10
              },
              "count": 5170,
              "dimensions": {
                "metric": "PTY"
              },
              "sum": {
                "edgeResponseBytes": 31573240,
                "visits": 3220
              }
            }
          ]
        }
      ]
    }
  },
  "errors": null
}
```

    ZapTimeseriesBydatetimeFifteenMinutesGroupedBycoloCode($zoneTag: string, $filter: ZoneHttpRequestsAdaptiveGroupsFilter_InputObject) {
  viewer {
    zones(filter: {zoneTag: $zoneTag}) {
      series: httpRequestsAdaptiveGroups(limit: 5000, filter: $filter) {
        count
        avg {
          sampleInterval
          __typename
        }
        sum {
          edgeResponseBytes
          visits
          __typename
        }
        dimensions {
          metric: coloCode
          ts: datetimeFifteenMinutes
        }
      }
    }
  }
}

```

This query says:

> - Given ???.

Author: Stole these few lines from one of the other migration guides - if we'd like to keep here, I'd appreciate some input on what goes here (Given a colo code, time range, and zone tag?)

Reviewer: Please find my comment above

- Fetch `count`, `sum(edgeResponseBytes)`, and `visits` colos analytics for `zoneTag` with a time range that starts on `2020-12-10T00:00:00Z`.

A few points to note:

> - Instead of `requests`, the `httpRequestsAdaptiveGroups` query reports `count`. This means number of requests per colocation.

Reviewer 3: I think the official Cloudflare term is "data center" and not "colocation" (which is weird too -- why not just say 'colo'?) -- but I think this is a call for @crwaters16 and the tech writing team.

Reviewer 3: Also @twalwyn this behavior is kind of weird now that we write it out. Should we add `requests` as an explicit field that can be counted?

> - The `httpRequestsAdaptiveGroups` query reports `sum(edgeResponseBytes)` instead of `bandwidth`. This is the same metric that `httpRequestsByColoGroups` reported as `bandwidth`.

Reviewer 3: I'm not sure about this one, but it might be worth being pedantic and pointing out that `bandwidth` wasn't really correct terminology for this field in the past.

> - `unique visitors per colocation` is not supported in `httpRequestsAdaptiveGroups`, but the `httpRequestsAdaptiveGroups` API does support `visits`. 

Reviewer 3: Again, would just say "data center" or at least 'colo'

Reviewer 3: Also, this guide should mention the distinction between a 'unique' and a 'visit'. 

Visits are defined here: https://support.cloudflare.com/hc/en-us/articles/360037684251-Understanding-Cloudflare-Site-Analytics#:~:text=Visits%20-%20A%20visit%20is%20defined%20as%20a%20page%20view%20that%20originated%20from%20a%20different%20website%2C%20or%20direct%20link.%20Cloudflare%20checks%20where%20the%20HTTP%20referer%20does%20not%20match%20the%20hostname.%20One%20visit%20can%20consist%20of%20multiple%20page%20views
